File: rb/features.rb
# WeatherForecast SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WeatherForecastFeatures
  def self.make_feature(name)
    case name
    when "base"
      WeatherForecastBaseFeature.new
    when "ratelimit"
      WeatherForecastRatelimitFeature.new
    when "retry"
      WeatherForecastRetryFeature.new
    when "test"
      WeatherForecastTestFeature.new
    when "timeout"
      WeatherForecastTimeoutFeature.new
    else
      WeatherForecastBaseFeature.new
    end
  end
end

File: py/features.py
# WeatherForecast SDK feature factory

from feature.base_feature import WeatherForecastBaseFeature
from feature.test_feature import WeatherForecastTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeatherForecastBaseFeature(),
        "test": lambda: WeatherForecastTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
